projects/dotcms-field-elements/src/components/dot-binary-file/dot-binary-file.tsx

```diff
@@ -0,0 +1,284 @@
+    URLValidationMessageWatch(): void {
+        this.errorMessageMap.set(DotBinaryMessageError.URLINVALID, this.URLValidationMessage);
+    }
+
+    @Watch('accept')
+    optionsWatch(): void {
+        this.accept = checkProp<DotBinaryFileComponent, string>(this, 'accept');
+        this.allowedFileTypes = !!this.accept ? this.accept.split(',') : [];
+    }
+
+    @Listen('fileChange')
+    fileChangeHandler(event: CustomEvent): void {
+        event.stopImmediatePropagation();
+        const fileEvent: DotBinaryFileEvent = event.detail;
+        this.errorType = fileEvent.errorType;
+        this.setValue(fileEvent.file);
+        if (this.isBinaryUploadButtonEvent(event.target as Element) && fileEvent.file) {
+            this.binaryTextField.value = (fileEvent.file as File).name;
+        }
+    }
+
+    @Listen('onBlur')
+    blurHandler(): void {
+        if (!this.status.dotTouched) {
+            this.status = updateStatus(this.status, {
+                dotTouched: true
+            });
+            this.emitStatusChange();
+        }
+    }
+
+    @Listen('dragover', { passive: false })
+    HandleDragover(evt: DragEvent): void {
+        evt.preventDefault();
+        if (!this.disabled) {
+            this.el.classList.add('dot-dragover');
+            this.el.classList.remove('dot-dropped');
+        }
+    }
+
+    @Listen('dragleave', { passive: false })
+    HandleDragleave(evt: DragEvent): void {
+        evt.preventDefault();
+        this.el.classList.remove('dot-dragover');
+        this.el.classList.remove('dot-dropped');
+    }
+
+    @Listen('drop', { passive: false })
+    HandleDrop(evt: DragEvent): void {
+        evt.preventDefault();
+        if (!this.disabled) {
+            this.el.classList.add('dot-dropped');
+            this.el.classList.remove('dot-dragover');
+            this.errorType = null;
+            const droppedFile: File = evt.dataTransfer.files[0];
+            if (isFileAllowed(droppedFile.name, this.allowedFileTypes)) {
+                this.setValue(droppedFile);
+                this.binaryTextField.value = droppedFile.name;
+            } else {
+                this.errorType = DotBinaryMessageError.INVALID;
+                this.setValue(null);
+            }
+        }
+    }
+
+    hostData() {
+        return {
+            class: getClassNames(this.status, this.isValid(), this.required)
+        };
+    }
+
+    render() {
+        return (
+            <Fragment>
+                <dot-label
+                    label={this.label}
+                    required={this.required}
+                    name={this.name}
+                    tabindex="0"
+                >
+                    <dot-binary-text-field
+                        placeholder={this.placeholder}
+                        required={this.required}
+                        disabled={this.disabled}
+                        accept={this.allowedFileTypes}
+                        hint={this.hint}
```

Comment:
```suggestion
                        hint={this.hint}
                        onLostFocus={this.blurHandler.bind(this)}
```

Reply:
to implement this we need to add a `prop` called, `onLostFocus` in `dot-binary-text-field`. think with the `@Listen` is more "natural" way to do it ?. The idea of catch the natural `blur` event in the parent don't work.

Reply:
Is not add, we already have an `onBlur` [here](https://github.com/dotCMS/core-web/pull/1069/files#diff-e6e8bbbffea72a93edb85d3eba36cc78R52) that when stencil compiles it creates a `onOnBlur` event, so is replace that.

I don't think `@Listen` is more natural because other devs can't know which components is the one that is throwing that event.